### PR TITLE
Read LAN Coordinates for a specific node

### PR DESCRIPTION
**Pull request by JocelynVelarde**
## Changes made
1. Added GET /v1/coordinate/node/+node into `Coordinate.cs`
2. Included new endpoints into `ICoordinateEndpoint.cs`
3. Added test `CoordinateTest.cs`

## Issue ticket number and link
Refers to issue #313

### CODE REVIEW

File: Consul/Interfaces/ICoordinateEndpoint.cs [hunk 1]
```diff
@@ -29,6 +29,7 @@ public interface ICoordinateEndpoint
     {
         Task<QueryResult<CoordinateDatacenterMap[]>> Datacenters(CancellationToken ct = default);
         Task<QueryResult<CoordinateEntry[]>> Nodes(CancellationToken ct = default);
+        Task<QueryResult<CoordinateEntry>> Node(string node, QueryOptions q, CancellationToken ct = default);
```
Comment (marcin-krystianc) [inline Consul/Interfaces/ICoordinateEndpoint.cs:32]: There are two overloads of `Nodes` endpoint (with and without `QueryOptions q` parameter).
We can Apply the same pattern for the new `Node` endpoint`

**Review by marcin-krystianc — COMMENTED** [review]
Tests are failing to compile (because of missing query options parameter). 
Please note also that the 'Node' endpoint returns an array (https://developer.hashicorp.com/consul/api-docs/v1.9.x/coordinate#sample-response-2), instead of a single object.
I'm not sure how to handle it. It seems unlikely that there will be more than one entry in the result, but since the endpoint returns an array we probably should use an array in the result as well.

File: Consul/Interfaces/ICoordinateEndpoint.cs (hunk 1 above)
Comment (JocelynVelarde) [reply Consul/Interfaces/ICoordinateEndpoint.cs:32]: Sure noted! Already fixed 🚀

**Comment by JocelynVelarde** [comment]
> Tests are failing to compile (because of missing query options parameter). Please note also that the 'Node' endpoint returns an array (https://developer.hashicorp.com/consul/api-docs/v1.9.x/coordinate#sample-response-2), instead of a single object. I'm not sure how to handle it. It seems unlikely that there will be more than one entry in the result, but since the endpoint returns an array we probably should use an array in the result as well.

I already fixed this on the recent commit, let me know if it needs further adjustments :)

**Comment by marcin-krystianc** [comment]
> > Tests are failing to compile (because of missing query options parameter). Please note also that the 'Node' endpoint returns an array (https://developer.hashicorp.com/consul/api-docs/v1.9.x/coordinate#sample-response-2), instead of a single object. I'm not sure how to handle it. It seems unlikely that there will be more than one entry in the result, but since the endpoint returns an array we probably should use an array in the result as well.
> 
> I already fixed this on the recent commit, let me know if it needs further adjustments :)

I've run the CI and there are some compilation issues. Please have a look.

**Comment by marcin-krystianc** [comment]
Hi @JocelynVelarde,

there is a problem with the `coordinates` endpoint, which returns empty responses for some time after starting the consul agent (e.g. https://github.com/G-Research/consuldotnet/actions/runs/8692709325/job/23862884528).
We can fix this by bumping the wait time and adding another check in the Base Fixture class:
image

**Comment by JocelynVelarde** [comment]
> Hi @JocelynVelarde,
> 
> 
> 
> there is a problem with the `coordinates` endpoint, which returns empty responses for some time after starting the consul agent (e.g. https://github.com/G-Research/consuldotnet/actions/runs/8692709325/job/23862884528).
> 
> We can fix this by bumping the wait time and adding another check in the Base Fixture class:
> 
> image
> 
> 
> 
> 

Alright noted! Is there any way I can test the endpoint locally (making the request) so I can check the output before submitting for review?

**Comment by marcin-krystianc** [comment]
> 
> Alright noted! Is there any way I can test the endpoint locally (making the request) so I can check the output before submitting for review?

Sure thing, there is a description how to to start local agent for tests -> https://consuldot.net/docs/getting-started/setting-up-server

**Comment by JocelynVelarde** [comment]
> 
> 
> > 
> 
> > Alright noted! Is there any way I can test the endpoint locally (making the request) so I can check the output before submitting for review?
> 
> 
> 
> Sure thing, there is a description how to to start local agent for tests -> https://consuldot.net/docs/getting-started/setting-up-server

Yeaaa! That's how I started running tests yesterday, so on the CLI I have to write the fetch request?

File: Consul/Coordinate.cs [hunk 2]
```diff
@@ -88,6 +88,23 @@ public Task<QueryResult<CoordinateEntry[]>> Nodes(QueryOptions q, CancellationTo
...
+        /// <param name="q">Customized query options</param>"
+        /// <remarks>Node is used to return the coordinates of a given node in the LAN pool.</returns>
+        public Task<QueryResult<CoordinateEntry[]>> Node(string node, QueryOptions q, CancellationToken ct = default)
+        {
+            return _client.Get<CoordinateEntry[]>(string.Format("/v1/coordinate/node/{0}", node), q).Execute(ct);
+        }
+
+        public Task<QueryResult<CoordinateEntry[]>> Node(string node, CancellationToken ct = default)
+        {
+            return _client.Get<CoordinateEntry[]>(string.Format("/v1/coordinate/node/{0}", node)).Execute(ct);
```
Comment (marcin-krystianc) [inline Consul/Coordinate.cs:106]: Can be replaced with:
```suggestion
            return Node(node, QueryOptions.Default, ct);
```

File: Consul.Test/BaseFixture.cs [hunk 3]
```diff
@@ -67,6 +67,10 @@ public class BaseFixture : IAsyncLifetime
                     // Workaround for https://github.com/hashicorp/consul/issues/15061
                     await client.Agent.GetAgentMetrics();
 
+                    /*var nodesResponse = await client.Coordinate.Nodes();
```
Comment (marcin-krystianc) [inline Consul.Test/BaseFixture.cs:70]: Is it commented out by mistake?

File: Consul.Test/CoordinateTest.cs [hunk 4]
```diff
@@ -49,5 +49,19 @@ public async Task Coordinate_GetNodes()
             // get an error. - from offical API.
             Assert.NotNull(nodes);
         }
+
+        [Fact]
+        public async Task Coordinate_GetNode()
+        {
```
Comment (marcin-krystianc) [inline Consul.Test/CoordinateTest.cs:55]: Previous version of the test was better, we did actually use a real node name in the query before.

File: Consul.Test/BaseFixture.cs (hunk 3 above)
Comment (JocelynVelarde) [reply Consul.Test/BaseFixture.cs:70]: Ohh let me activate it again

File: Consul/Coordinate.cs (hunk 2 above)
Comment (JocelynVelarde) [reply Consul/Coordinate.cs:106]: Alright done

File: Consul.Test/CoordinateTest.cs (hunk 4 above)
Comment (JocelynVelarde) [reply Consul.Test/CoordinateTest.cs:55]: The thing is that with the node name I was receiving errors in the base fixture class

**Review by marcin-krystianc — COMMENTED** [review]
Thanks, looks good. Just one final remark about removing unnecessary code.

**Review by marcin-krystianc — APPROVED** [review]
Now it is perfect, thank you!
I need to sort out macOS CI failures before merging it.

**Pull request merged**